inst/r_interface/model_output.hpp.in

```diff
@@ -0,0 +1,27 @@
+{{header}}
+
+#pragma once
+
+#include <Rcpp.h>
+#include <unsupported/Eigen/CXX11/Tensor>
+
+#include "state_saver.hpp"
+
+template<leapfrog::HivAgeStratification S, typename real_type>
+Rcpp::List build_r_output(const typename leapfrog::StateSaver<S, real_type>::OutputState &state,
+                          const std::vector<int> save_steps) {
+  constexpr auto ss = leapfrog::StateSpace<S>();
```

Comment:
how do we keep track of these being the variables to export?

Reply:
I wonder if you could keep a vector of what is in `StateSpace` and unpack them all in the generated code, or intersect them with things found in the csv?

Reply:
This is just manual at the moment. Yeah really I think we can unpack everything in the state space. I want to ultimately have these defined in 1 place. Might look at if we can expose a function from C++ which this generation can call to list all available state space variables which we can use.

Reply:
Actually looks like that kind of introspection is not possible. But this would be much simpler if all of these dimensions came from the state space. Slightly complicated by the fact that `age_groups_fert` and `hiv_adult_first_age_group` come from the options. If we moved those into the state space we could then generate this from if it is used in the csv, assume from state space and generate a line to unpack it like this.

Reply:
Ok will put this into a little helper function for now, but will try revisit after I have the next stage of code generation in so that if we do extend it we only have to add it in 1 place